requirements/tests.txt

```diff
@@ -4,4 +4,6 @@ pytest-benchmark
 pytest-cov
 mypy
 black
-darglint
\ No newline at end of file
+darglint
+tensorflow_datasets
```

Comment:
We might want to move these to the optional requirements similarly to pytorch

Reply:
Added the conditional check but want to have these to requirements/tests so that we get all requirements to run all tests.

Reply:
@imshashank I don't think that's true. We have the `simple` and `complex` test suites for precisely this reason. The simple tests won't run with optional dependencies; this seems like the exact use-case for it.

Reply:
Updated